common/src/main/java/com/wynntils/core/net/NetManager.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.net;
+
+import com.google.gson.JsonObject;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.util.Map;
+import net.minecraft.Util;
+import org.apache.commons.codec.digest.DigestUtils;
+
+public class NetManager extends CoreManager {
+    protected static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+
+    private static final int REQUEST_TIMEOUT_MILLIS = 10000;
```

Comment:
In the future, I might PR this to be an exponential timeout. So we timeout on 2 second, retry 4 seconds, then 8. That might be better overall.

Reply:
Maaaybe..? Because aborting and retrying again would be better than waiting for to long? Right now we do not do any reconnection retries. (Neither did the old code). That might be a point.

Overall, we do not handle very well if network downloading is shaky. I think this PR puts the foundation for a more robust approach, but to actually implement proper handling in the face of network errors will require more thorough changes to the entire code base. And serious testing of adverse situations, otherwise the code is likely to break anyway, if it never has been tested.